lib/catalog/order_item_sanitized_parameters.rb

```diff
@@ -37,18 +37,13 @@ def service_parameters_raw
     end
 
     def filtered_parameters
-      params = service_parameters_raw.slice(*fields.collect { |field| field.with_indifferent_access["name"] })
-      params.transform_values! { |v| substitute(v) }
+      params = service_parameters_raw.slice(*fields.collect { |field| field[:name] })
+      params.collect { |key, value| [key, sanitize_value(key, value)] }.to_h
     end
 
     def service_plan_schema
-      TopologicalInventory::Service.call do |api|
-        @plan = api.show_service_plan(service_plan_ref.to_s)
-      end
-      @plan.create_json_schema
-    rescue ::Catalog::TopologyError => e
-      Rails.logger.error("DefaultApi->show_service_plan #{e.message}")
-      raise
+      service_plan = order_item.portfolio_item.service_plans.first
```

Comment:
@bzwei Should we check if the service plan is similar to the one in the tower there were some changes made by @lindgrenj6 to keep the 2 schemas in sync. If we have an old schema in catalog the order might fail. I am not sure how we are tracking that.

Reply:
it is already checked before this class is called. https://github.com/RedHatInsights/catalog-api/blob/master/lib/catalog/submit_order_item.rb#L12